lib/ansible/vault/key_value_decryptor.rb

```diff
@@ -0,0 +1,36 @@
+require 'yaml'
+require 'ansible/vault/util'
+
+module Ansible
+  class Vault
+    # The module that decrypting vault key-value pair(s)
+    module KeyValueDecryptor
+      # Decrypts the key-value pair(s)
+      #
+      # @param text [String] The encrypted text
+      # @param password [String] The password for the text
+      # @return [Map<String>String] The key(s) and plaintext contents of the ciphertext.
+      def decrypt(text:, password:)
+        # TODO: `safe_load` usage is better, but how much classes to whitelist?
+        result = YAML.load(text)
```

Comment:
This is another one open for suggestions. @sunakan @tpickett66 

The API contract of `#decrypt(text:,password:)` is kept for the moment.

However if we were to use `YAML.safe_load`, the end user would surely want to be able to decide which classes to whitelist.

This won't be possible with the current API contract.

Then we get to the scenario where we can change the contract to `Ansible::Vault::KeyValueDecryptor.decrypt(text:, password:, yaml_whitelist_classes:)` and call it a day, however what would the usage be at `Ansible::Vault.decrypt(text:,password:)`? 

We can provide default white-listed classes and still have the developer go to `Ansible::Vault::KeyValueDecryptor.decrypt` if he/she wants to configure the whitelisted classes

Reply:
tl;dr I like this direction. I'll update the PR.

Hmm.. I haven't thought in the way of defining a custom scalar, but your guess was correct.

This is what Ansible does.

```
AnsibleConstructor.add_constructor(
    u'!vault',
    AnsibleConstructor.construct_vault_encrypted_unicode)
```

ref: https://github.com/ansible/ansible/blob/devel/lib/ansible/parsing/yaml/constructor.py#L154

and a bit more relevant info


```
# Unicode like object that is not evaluated (decrypted) until it needs to be
# TODO: is there a reason these objects are subclasses for YAMLObject?
class AnsibleVaultEncryptedUnicode(yaml.YAMLObject, AnsibleBaseYAMLObject):
    __UNSAFE__ = True
    __ENCRYPTED__ = True
    yaml_tag = u'!vault'

    @classmethod
    def from_plaintext(cls, seq, vault, secret):
        if not vault:
            raise vault.AnsibleVaultError('Error creating AnsibleVaultEncryptedUnicode, invalid vault (%s) provided' % vault)

        ciphertext = vault.encrypt(seq, secret)
        avu = cls(ciphertext)
        avu.vault = vault
        return avu

    def __init__(self, ciphertext):
        '''A AnsibleUnicode with a Vault attribute that can decrypt it.

        ciphertext is a byte string (str on PY2, bytestring on PY3).

        The .data attribute is a property that returns the decrypted plaintext
        of the ciphertext as a PY2 unicode or PY3 string object.
        '''
        super(AnsibleVaultEncryptedUnicode, self).__init__()

        # after construction, calling code has to set the .vault attribute to a vaultlib object
        self.vault = None
        self._ciphertext = to_bytes(ciphertext)

    @property
    def data(self):
        if not self.vault:
            # FIXME: raise exception?
            return self._ciphertext
        return to_text(self.vault.decrypt(self._ciphertext))
```

ref: https://github.com/ansible/ansible/blob/devel/lib/ansible/parsing/yaml/objects.py#L73